### PR TITLE
Issue 16984 pp default actions

### PR DESCRIPTION
**Pull request by jdotcms**

### CODE REVIEW

File: dotCMS/src/integration-test/java/com/dotcms/workflow/helper/TestSystemActionMappingsHandlerMerger.java [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+import com.dotmarketing.portlets.workflows.business.WorkflowAPI.SystemAction;
+import com.dotmarketing.portlets.workflows.model.SystemActionWorkflowActionMapping;
+import com.dotmarketing.portlets.workflows.model.WorkflowAction;
+import com.dotmarketing.portlets.workflows.model.WorkflowScheme;
+import com.dotmarketing.portlets.workflows.model.WorkflowStep;
+import com.dotmarketing.util.UUIDUtil;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.UUID;
```
[the remark this answers is not in the document]
Comment (jdotcms) [reply dotCMS/src/integration-test/java/com/dotcms/workflow/helper/TestSystemActionMappingsHandlerMerger.java]: Done

File: dotCMS/src/integration-test/java/com/dotcms/workflow/helper/TestSystemActionMappingsHandlerMerger.java [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+import org.junit.Test;
+
+/**
+ * This test is for handling the merging between system action mappings for content types and schemes
+ * @author jsanca
+ */
+public class TestSystemActionMappingsHandlerMerger extends IntegrationTestBase {
+
+    @BeforeClass
+    public static void prepare() throws Exception {
```
[the remark this answers is not in the document]
Comment (jdotcms) [reply dotCMS/src/integration-test/java/com/dotcms/workflow/helper/TestSystemActionMappingsHandlerMerger.java]: nvm

File: dotCMS/src/main/java/com/dotcms/publisher/pusher/wrapper/WorkflowWrapper.java [hunk 3]
```diff
@@ -21,19 +20,36 @@
...
-		this.actionStepsListMap = actionStepsListMap;
+		this.actionStepsListMap         = actionStepsListMap;
+		this.systemActionMappings       = systemActionMappings;
 	}
 
+	public List<SystemActionWorkflowActionMapping> getSystemActionMappings() {
+		return systemActionMappings;
+	}
+
+	public void setSystemActionMappings(List<SystemActionWorkflowActionMapping> systemActionMappings) {
```
[the remark this answers is not in the document]
Comment (jdotcms) [reply dotCMS/src/main/java/com/dotcms/publisher/pusher/wrapper/WorkflowWrapper.java]: Done

File: dotCMS/src/main/java/com/dotcms/workflow/helper/SystemActionMappingsHandlerMerger.java [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
+                }
+            }
+        }
+    }
+
+    private Optional<WorkflowAction> findWorkflowAction (final String workflowId) {
+
+        try {
+            return Optional.ofNullable(this.workflowAPI.findAction(workflowId, APILocator.systemUser()));
+        } catch (Exception e) {
```
[the remark this answers is not in the document]
Comment (jdotcms) [reply dotCMS/src/main/java/com/dotcms/workflow/helper/SystemActionMappingsHandlerMerger.java]: nvm

File: dotCMS/src/main/java/com/dotmarketing/portlets/workflows/model/SystemActionWorkflowActionMappingDeserializer.java [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+            owner = readValueAs(objectNode.get("owner"), WorkflowScheme .class, parser);
+        }
+
+        return new SystemActionWorkflowActionMapping(identifier, systemAction, workflowAction, owner);
+    }
+
+    private ContentType readContentTypeValueAs(final JsonNode node) {
+
+        final String id     = node.get("id").asText();
```
[the remark this answers is not in the document]
Comment (jdotcms) [reply dotCMS/src/main/java/com/dotmarketing/portlets/workflows/model/SystemActionWorkflowActionMappingDeserializer.java]: it is ok

File: dotCMS/src/main/java/com/dotmarketing/portlets/workflows/model/WorkflowScheme.java [hunk 6]
```diff
@@ -136,7 +137,18 @@ public String entryStep() {
         return this.entryStep;
 
     }
-	
+
+	@Override
+	public boolean equals(Object obj) {
```
[the remark this answers is not in the document]
Comment (jdotcms) [reply dotCMS/src/main/java/com/dotmarketing/portlets/workflows/model/WorkflowScheme.java]: Done

File: dotCMS/src/main/java/com/dotmarketing/portlets/workflows/model/WorkflowScheme.java [hunk 7]
```diff
@@ -136,7 +137,18 @@ public String entryStep() {
         return this.entryStep;
 
     }
-	
+
+	@Override
+	public boolean equals(Object obj) {
+		if(obj ==null || ! (obj instanceof WorkflowScheme)) return false;
```
[the remark this answers is not in the document]
Comment (jdotcms) [reply dotCMS/src/main/java/com/dotmarketing/portlets/workflows/model/WorkflowScheme.java]: Done

File: dotCMS/src/main/java/com/dotmarketing/portlets/workflows/util/WorkflowImportExportUtil.java [hunk 8]
```diff
@@ -35,6 +35,7 @@
 import java.util.Map;
 import java.util.Set;
 import java.util.stream.Collectors;
+import javax.ws.rs.HEAD;
```
[the remark this answers is not in the document]
Comment (jdotcms) [reply dotCMS/src/main/java/com/dotmarketing/portlets/workflows/util/WorkflowImportExportUtil.java]: Done

**Review by jgambarios — APPROVED** [review]

**Pull request merged**